exp/runtime/hooks/api/v1alpha1/common_types.go

```diff
@@ -16,6 +16,46 @@ limitations under the License.
 
 package v1alpha1
 
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Response can set and get fields common to all Response types.
+// +kubebuilder:object:generate=false
+type Response interface {
+	runtime.Object
+	GetMessage() string
+	GetStatus() ResponseStatus
+	SetMessage(message string)
+	SetStatus(status ResponseStatus)
+}
+
+// CommonResponse is the data structure common to all Response types.
+type CommonResponse struct {
```

Comment:
What about
```suggestion
// ResponseMeta describes metadata that all response messages must have, including status
// and error message.
type ResponseMeta struct {
```

Reply:
Is this really metadata?

Reply:
I'm also not sure how we can build on top of this naming when we add another struct to cover the RetryAfterSeconds field.

Reply:
We would prefer CommonResponse as those fields feel more like fields common to Responses.

They only tell us if the response as a whole is success/failure, but apart from that they don't tell us anything about other attributes of the response (so meta doesn't seem to fit well)